source/s3_auto_ranged_get.c

```diff
@@ -651,11 +699,16 @@ static int s_discover_object_range_and_content_length(
 
                 break;
             }
-
-            /* When discovering the object size via first-part, the object range is the entire object. */
-            object_range_start = 0;
-            object_range_end = total_content_length - 1; /* range-end is inclusive */
-
+            if (auto_ranged_get->initial_message_has_range_header) {
+                object_range_end = object_size - 1;
+                if (auto_ranged_get->initial_message_has_end_range) {
+                    object_range_end = aws_min_u64(object_range_end, auto_ranged_get->initial_object_range_end);
+                }
+            } else {
+                /* When discovering the object size via first-part, the object range is the entire object. */
```

Comment:
utterly trivial: this is probably an old comment, but "via first-part" is a bit confusing, now that we have GET_OBJECT_WITH_PART_NUMBER_1, but this comment is referring to first-part-with-range

Reply:
Thanks, updated.